GVFS/GVFS.Common/Http/HttpRequestor.cs

```diff
@@ -249,10 +316,69 @@ private static string GetSingleHeaderOrEmpty(HttpHeaders headers, string headerN
             IEnumerable<string> values;
             if (headers.TryGetValues(headerName, out values))
             {
-                return values.First();                
+                return values.First();
             }
 
             return string.Empty;
         }
+
+        private string LoadCertificatePassword(string certId, GitProcess git)
+        {
+            if (git.TryGetCertificatePassword(this.Tracer, certId, out var password, out var error))
+            {
+                return password;
+            }
+
+            return null;
+        }
+
+        private X509Certificate2 LoadCertificate(string certId, string certificatePassword, bool onlyLoadValidCertificateFromStore)
+        {
+            EventMetadata metadata = new EventMetadata
+            {
+                { "certId", certId },
+                { "isPasswordSpecified", string.IsNullOrEmpty(certificatePassword) },
+                { "shouldVerify", onlyLoadValidCertificateFromStore }
+            };
+
+            if (File.Exists(certId))

```

Comment:
What is the purpose of this check?  Null or empty was already checked before this method was called and the file could be deleted between this call and the constructor for `X509Certificate2` and you are already handling the exception for that but it will be a different code path.  

If the file doesn't exist you run `Find`. If it was deleted between this call and the `X509Certificate2` constructor which means it doesn't exist it will return null.  Not sure how much this matters, just pointing out the difference in code paths.

Reply:
There might be a way to parse the string and determine if the format is a path or name but that seems overly complex so this check will be fine.

Reply:
It's too complex, and you can't determine, if a string "certificate" refers to a file in your current working directory, or to a subject name in a certificate store.